src/utils.py

```diff
@@ -159,8 +159,12 @@ def convert_version_to_proper_semantic(version):
 
 
 def select_latest_version(input_version='', libio='', anitya=''):
-    libio_latest_version = convert_version_to_proper_semantic(libio)
-    anitya_latest_version = convert_version_to_proper_semantic(anitya)
+    libio_latest_version = '0.0.0'
+    anitya_latest_version = '0.0.0'
+    if libio and libio != "-1":
```

Comment:
@miteshvp Correct me if I am wrong, you want if `libio` or `anitya` is `-1` you want it's latest_version to be `0.0.0` instead of `' '` but the `convert_version_to_proper_semantic()` function does handle that case here https://github.com/fabric8-analytics/f8a-server-backbone/blob/master/src/utils.py#L155

Reply:
cc @ravsa

Reply:
I guess I figured out what the problem is in the https://github.com/fabric8-analytics/f8a-server-backbone/blob/master/src/utils.py#L155 line.
It needs to be `version in ('', '-1',None)`, currently is it just -1, which is being treated as an int and not string, and we never caught this error as there are no strict type checking in Python.

Reply:
yes, correct @sara-02  and we should add this check above https://github.com/fabric8-analytics/f8a-server-backbone/pull/34/files#diff-e4e95f0c8626ae81416f8fa5477fbfa1R154

Reply:
Yes, it makes sense, will send a PR with the fix.Thanks